codeinsights-db/conf/postgresql.conf

```diff
@@ -0,0 +1,781 @@
+					# encoding
+
+# These settings are initialized by initdb, but they can be changed.
+lc_messages = 'en_US.utf8'			# locale for system error message
+					# strings
+lc_monetary = 'en_US.utf8'			# locale for monetary formatting
+lc_numeric = 'en_US.utf8'			# locale for number formatting
+lc_time = 'en_US.utf8'				# locale for time formatting
+
+# default configuration for text search
+default_text_search_config = 'pg_catalog.english'
+
+# - Shared Library Preloading -
+
+#shared_preload_libraries = ''	# (change requires restart)
+#local_preload_libraries = ''
+#session_preload_libraries = ''
+#jit_provider = 'llvmjit'		# JIT library to use
+
+# - Other Defaults -
+
+#dynamic_library_path = '$libdir'
+
+
+#------------------------------------------------------------------------------
+# LOCK MANAGEMENT
+#------------------------------------------------------------------------------
+
+#deadlock_timeout = 1s
+#max_locks_per_transaction = 64		# min 10
+					# (change requires restart)
+#max_pred_locks_per_transaction = 64	# min 10
+					# (change requires restart)
+#max_pred_locks_per_relation = -2	# negative values mean
+					# (max_pred_locks_per_transaction
+					#  / -max_pred_locks_per_relation) - 1
+#max_pred_locks_per_page = 2            # min 0
+
+
+#------------------------------------------------------------------------------
+# VERSION AND PLATFORM COMPATIBILITY
+#------------------------------------------------------------------------------
+
+# - Previous PostgreSQL Versions -
+
+#array_nulls = on
+#backslash_quote = safe_encoding	# on, off, or safe_encoding
+#escape_string_warning = on
+#lo_compat_privileges = off
+#operator_precedence_warning = off
+#quote_all_identifiers = off
+#standard_conforming_strings = on
+#synchronize_seqscans = on
+
+# - Other Platforms and Clients -
+
+#transform_null_equals = off
+
+
+#------------------------------------------------------------------------------
+# ERROR HANDLING
+#------------------------------------------------------------------------------
+
+#exit_on_error = off			# terminate session on any error?
+#restart_after_crash = on		# reinitialize after backend crash?
+#data_sync_retry = off			# retry or panic on failure to fsync
+					# data?
+					# (change requires restart)
+
+
+#------------------------------------------------------------------------------
+# CONFIG FILE INCLUDES
+#------------------------------------------------------------------------------
+
+# These options allow settings to be loaded from files other than the
+# default postgresql.conf.  Note that these are directives, not variable
+# assignments, so they can usefully be given more than once.
+
+#include_dir = '...'			# include files ending in '.conf' from
+					# a directory, e.g., 'conf.d'
+#include_if_exists = '...'		# include file only if it exists
+#include = '...'			# include file
+
+
+#------------------------------------------------------------------------------
```

Comment:
> Do you think we could get rid of everything above our customizations? Those should all be the defaults and not required to be in this file (I think). That would make our changes much more visible for customers instead of being at the end of a 800-line file

It looks like we can't. The default `listen_address` is localhost (https://postgresqlco.nf/doc/en/param/listen_addresses/), the migrator (or any other services) will never able to reach pgsql.

The right way would have been do not touch the default postgresql.conf, but only insert a `include_dir '/conf'` at the end. By doing so, PG can load all conf files during runtime.

I am debating whether this is something we want to do, feels like a big change since we may need to change the postgres image https://github.com/sourcegraph/sourcegraph/tree/main/docker-images/postgres-12-alpine

thoughts?

Reply:
Ugh, yeah. There's about 10 settings that are customized from the defaults hidden away in that file. I still think it would be beneficial to remove all the commented-out sections and only leave the things that are being explicitly set.

Your point about the `include_dir` is good, but it does feel like a pretty large task with some risk of losing customer overrides.

How do you feel about removing just the commented out sections?

Reply:
Ugh, yeah. There's about 10 settings that are customized from the defaults hidden away in that file. I still think it would be beneficial to remove all the commented-out sections and only leave the things that are being explicitly set.

Your point about the `include_dir` is good, but it does feel like a pretty large task with some risk of losing customer overrides.

How do you feel about removing just the commented out sections?